### PR TITLE
book controller implementation

### PR DESCRIPTION
**Pull request by c-kalenyk**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -11,14 +15,25 @@
...
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't get book by id: " + id));
+        return bookMapper.toBookDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toBookDto)
+                .toList();
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java]: FYI you can declare mapping list to list in mapstruct inteface

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.bookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java]: it's ok to use Data on DTOs, but not on JPA entities

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/books")
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/controller/BookController.java]: /api will be common for all controllers - extract it to application.properties 
server.servlet.context-path

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]
Well done

**Pull request merged**